poly/src/domain/radix2/fft.rs

```diff
@@ -141,94 +141,160 @@ impl<F: FftField> Radix2EvaluationDomain<F> {
         // In the sequential case, we will keep on making the roots cache-aligned,
         // according to the access pattern that the FFT uses.
         // It is left as a TODO to implement this for the parallel case
-        let roots = &mut self.roots_of_unity(root);
+        let mut roots = self.roots_of_unity(root);
+
+        #[cfg(feature = "parallel")]
+        let max_threads = rayon::current_num_threads();
 
         #[cfg(not(feature = "parallel"))]
-        let mut root_len = roots.len();
+        let max_threads = 1;
 
         let mut gap = xi.len() / 2;
         while gap > 0 {
             // each butterfly cluster uses 2*gap positions
             let chunk_size = 2 * gap;
-            #[cfg(feature = "parallel")]
-            let nchunks = xi.len() / chunk_size;
+            let num_chunks = xi.len() / chunk_size;
+            let sub_chunks = (max_threads - 1) / num_chunks + 1;
+            let sub_chunk_size = chunk_size / (2 * sub_chunks);
```

Comment:
Could you add some comments on what's happening here? In particular, why does this choice of constants work?